python310Packages.pyasn1-modules: 0.2.8 -> 0.3.0, python310Packages.pyasn1: 0.4.8 -> 0.5.0

fabaff opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

fabaff commented:
Result of `nixpkgs-review pr 227630` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

SuperSandro2000 commented on pkgs/development/python-modules/pyasn1-modules/default.nix:
```diff
@@ -1,35 +1,43 @@
     license = licenses.bsd3;
-    platforms = platforms.unix;  # same as pyasn1
+    maintainers = with maintainers; [ ];
+    platforms = platforms.unix;
```
```suggestion
```
default for buildPythonPackage

SuperSandro2000 commented on pkgs/development/python-modules/pyasn1/default.nix:
```diff
@@ -1,19 +1,29 @@
     description = "Generic ASN.1 library for Python";
-    homepage = "https://github.com/etingof/pyasn1";
+    homepage = "https://pyasn1.readthedocs.io";
+    chaangelog = "https://github.com/etingof/pyasn1/blob/master/CHANGES.rst";
```
```suggestion
    changelog = "https://github.com/etingof/pyasn1/blob/master/CHANGES.rst";
```

SuperSandro2000 commented:
We don't need to split the commits that much, three would already be enough IMO

fabaff committed:
python310Packages.pyasn1-modules: 0.2.8 -> 0.3.0

Diff: pyasn1/pyasn1-modules@refs/tags/v0.2.8...v0.3.0

Changelog: https://github.com/pyasn1/pyasn1-modules/releases/tag/v0.3.0

fabaff committed:
python310Packages.pyasn1: 0.4.8 -> 0.5.0

Changelog: https://github.com/etingof/pyasn1/blob/master/CHANGES.rst

fabaff commented:
Needs to go to `staging`...

vcunat commented:
This broke tests of `python3*Packages.twisted`, and through that hundreds of other builds.  aarch64-linux log: https://hydra.nixos.org/build/219423199/nixlog/1/tail

Note that we are (and were) in period where "breaking changes" are disallowed.  Though with python packages it's often hard to tell in advance, I suppose.  [GitHub tag](https://github.com/pyasn1/pyasn1/releases/tag/v0.5.0) calls it a "major release", so maybe it's not suitable for 23.05 anymore.  But I guess it might be OK to keep, if _someone_ fixes the regressions quickly, etc.

SuperSandro2000 commented:
#231253

mweinelt commented:
Also broke certipy.

https://hydra.nixos.org/build/219691949
